### ISSUE TITLE
[ci][instrumentation-mongodb] tav tests are failing

### ISSUE THREAD

**Issue by pichlermarc** [open-telemetry/opentelemetry-js-contrib#2309]
Tests for `instrumentation-mongodb` are currently failing.

See https://github.com/open-telemetry/opentelemetry-js-contrib/actions/runs/9709436608/job/26798116241.

**Comment by pichlermarc** [comment]
Looks like the problem is `mongodb@6.8.0` that was released 18 hours ago (see https://github.com/mongodb/node-mongodb-native/releases/tag/v6.8.0)

I assume it to be a side effect of "Cursor responses are now parsed lazily"

The error from the tests:

```
2024-06-28T09:28:20.2247855Z      MongoRuntimeError: Unexpected null cursor id. A cursor creating command should have set this
2024-06-28T09:28:20.2249200Z       at FindCursor.getMore (node_modules/mongodb/src/cursor/abstract_cursor.ts:652:13)
2024-06-28T09:28:20.2250498Z       at FindCursor.getMore (node_modules/mongodb/src/cursor/find_cursor.ts:107:34)
2024-06-28T09:28:20.2251796Z       at FindCursor.fetchBatch (node_modules/mongodb/src/cursor/abstract_cursor.ts:730:35)
2024-06-28T09:28:20.2253263Z       at processTicksAndRejections (node:internal/process/task_queues:96:5)
2024-06-28T09:28:20.2254587Z       at FindCursor.next (node_modules/mongodb/src/cursor/abstract_cursor.ts:398:7)
2024-06-28T09:28:20.2255966Z       at FindCursor.[Symbol.asyncIterator] (node_modules/mongodb/src/cursor/abstract_cursor.ts:319:26)
2024-06-28T09:28:20.2257359Z       at FindCursor.toArray (node_modules/mongodb/src/cursor/abstract_cursor.ts:461:22)
```

**Comment by pichlermarc** [comment]
Bumping to P1 as it's causing problems in end-user apps.

**Comment by david-luna** [comment]
@pichlermarc indeed there is an internal change in the feature you mentioned which the instrumentation does not handle and raises the error
https://github.com/mongodb/node-mongodb-native/pull/4112/files#diff-588dd05062f39c3e44bcc04d3fe0ca9b370bdfb7469e4ce02d568373b2685cae

`Connection#command` now accepts an extra parameter and instrumentation does not pass it to the original function.

**Comment by pichlermarc** [comment]
Hmm, properly passing it makes the test pass, but there's still another test (`should attach response hook data to the resulting span for find function`) that's failing. I don't think that test's assertion may be faulty for that version though (IIUC it asserts on a JSON stringified _id field added by a custom hook, and that _id property is not there anymore, but the hook is working correctly).

**Comment by david-luna** [comment]
The hook is inspecting data from the response object but with this new `lazy parsing` feature operations with cursors like `find` won't have the data available until accessed via the cursor API.

- so maybe is not the best way to test the hook. Maybe we should change the hook?
- if anyone is using that hook to inspect data it won't work for them when upgrading to v6.8.0. Should we add a note?

**Comment by david-luna** [comment]
@pichlermarc I didn't mean to step on your PR but I already had a branch in the works when I commented in the issue. 
Here's the PR https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2314

**Comment by pichlermarc** [comment]
> @pichlermarc I didn't mean to step on your PR but I already had a branch in the works when I commented in the issue. Here's the PR #2314

No problem at all, thanks for working on this. My draft was more there to illustrate the problems I had. 
I appreciate you picking this up. :slightly_smiling_face:

**Comment by david-luna** [comment]
fixed in #2314
